src/riscv/lib/src/state_backend/proof_backend.rs

```diff
@@ -288,6 +358,16 @@ pub struct ProofDynRegion<const LEN: usize, M: ManagerBase> {
     writes: BTreeMap<usize, u8>,
 }
 
+impl<const LEN: usize, M: ManagerClone> Clone for ProofDynRegion<LEN, M> {
+    fn clone(&self) -> Self {
+        Self {
+            source: M::clone_dyn_region(&self.source),
+            reads: self.reads.clone(),
+            writes: self.writes.clone(),
+        }
```

Comment:
Would you mind placing this impl after `impl _` blocks? I would like the code base to be consistent wrt to the ordering: type definition, then `impl _`, then `impl _ for _`.

Reply:
I remember the convention was swapped for the last two types of impl blocks. (e.g. `PagePermissions`, `registers.rs`) Happy to change. [diff](https://github.com/tezos/riscv-pvm/compare/4da9f9f47bd55af81cd3bb785fc8b0b8ca752230..d48daa94bdcf3b97bb1644145226f8aff4c8ee2e)

Reply:
I'm not sure what you mean by `PagePermissions`; it seems to be in line with the ordering. The `registers` module is all over the place, indeed.